gtsam/hybrid/tests/testDCSAM.cpp

```diff
@@ -0,0 +1,685 @@
+
+  hfg.push_back(dcMixture);
+
+  // Let's make an initial guess
+  Values initialGuess;
+  double initVal = -2.5;
+  initialGuess.insert(x1, initVal);
+
+  // We also need an initial guess for the discrete variables
+  // (this will only be used if it is needed by your factors),
+  // here it is ignored.
+  DiscreteValues initialGuessDiscrete;
+  initialGuessDiscrete[dk.first] = 0;
+
+  // Let's make a solver
+  DCSAM dcsam;
+
+  // Add the HybridFactorGraph to DCSAM
+  dcsam.update(hfg, initialGuess, initialGuessDiscrete);
+
+  // Solve
+  HybridValues dcvals = dcsam.calculateEstimate();
+
+  // Run another iteration
+  dcsam.update();
+
+  // Update HybridValues
+  dcvals = dcsam.calculateEstimate();
+
+  // Ensure that the prediction is correct
+  const size_t mpeD = dcvals.discrete().at(dk.first);
+  EXPECT_LONGS_EQUAL(0, mpeD);
+}
+
+namespace slam_fixture {
+const double prior_sigma = 0.1;
+const double meas_sigma = 1.0;
+
+auto prior_noise = noiseModel::Isotropic::Sigma(3, prior_sigma);
+auto meas_noise = noiseModel::Isotropic::Sigma(3, meas_sigma);
+
+Pose2 pose0(0, 0, 0);
+Pose2 dx(1, 0, 0.78539816);
+Pose2 noise(0.01, 0.01, 0.01);
+
+}  // namespace slam_fixture
+
+/**
+ * This is a basic (qualitative) octagonal pose graph SLAM test
+ * to verify that DCSAM works on standard SLAM examples.
+ */
+TEST(DCSAM, SimpleSlamBatch) {
+  using namespace slam_fixture;
+
+  // Make a hybrid factor graph
+  HybridNonlinearFactorGraph graph;
+
+  // Values for initial guess
+  Values initialGuess;
+
+  Key x0 = X(0);
+
+  PriorFactor<Pose2> p0(x0, pose0, prior_noise);
+  graph.push_back(p0);
+
+  initialGuess.insert(x0, pose0);
+
+  // Setup dcsam
+  DCSAM dcsam;
+
+  Pose2 odom(pose0);
+  for (size_t i = 0; i < 7; i++) {
+    Key xi = X(i);
+    Key xj = X(i + 1);
+
+    Pose2 meas = dx * noise;
+
+    BetweenFactor<Pose2> bw(xi, xj, dx * noise, meas_noise);
+    graph.push_back(bw);
+
+    odom = odom * meas;
+    initialGuess.insert(xj, odom);
+  }
+
+  BetweenFactor<Pose2> bw(x0, X(7), dx * noise, meas_noise);
+
+  dcsam.update(graph, initialGuess);
+  HybridValues dcvals = dcsam.calculateEstimate();
+
+  // Values from the DCSAM repo
+  Values expectedContinuous;
+  expectedContinuous.insert<Pose2>(
+      X(0), Pose2(-3.53708e-33, -8.97559e-34, 4.1955e-33));
+  expectedContinuous.insert<Pose2>(X(1), Pose2(1, 0.0141421, 0.795398));
+  expectedContinuous.insert<Pose2>(X(2), Pose2(1.6899, 0.738184, 1.5908));
+  expectedContinuous.insert<Pose2>(X(3), Pose2(1.65576, 1.7377, 2.38619));
+  expectedContinuous.insert<Pose2>(X(4), Pose2(0.918069, 2.41298, -3.10159));
+  expectedContinuous.insert<Pose2>(X(5), Pose2(-0.0805657, 2.35886, -2.30619));
+  expectedContinuous.insert<Pose2>(X(6), Pose2(-0.740961, 1.60781, -1.5108));
+  expectedContinuous.insert<Pose2>(X(7), Pose2(-0.66688, 0.61046, -0.715398));
+
+  EXPECT(assert_equal(expectedContinuous, dcvals.nonlinear(), 1e-5));
+}
+
+/**
+ * This is a basic qualitative octagonal pose graph SLAM test
```

Comment:
```suggestion
 * This is a basic octagonal pose graph SLAM test
```
(as above)

It's worth mentioning that both of these tests are really just making sure the continuous solver in DCSAM does what we expect. They're not suuuper valuable, but it is nice to know that running the continuous solver inside DC-SAM isn't fundamentally breaking things, I guess.